### ISSUE TITLE
Migrate to null safety

### ISSUE THREAD

**Issue by GiorgiBeriashvili** [lomsa-dev/http-mock-adapter#67]
## Description

Dart has recently obtained the ability to enable sound null safety in Dart code. [Here](https://dart.dev/null-safety) is the overview of the feature. It is currently in beta.

From the [migration guide](https://dart.dev/null-safety/migration-guide) we see the following notice:

> **Migrating an app is technically the same as migrating a package.** Before migrating an app, consider waiting until null safety is in a stable release and all your dependencies are ready.

This means that it's generally recommended to migrate if developing a Dart package. Thus, it would seem beneficial to implement null safety features in `http_mock_adapter`.

**Comment by kuhnroyal** [comment]
Dio now has a `4.0.0-beta2` available with null-safety. It is no longer compatible.

**Comment by kuhnroyal** [comment]
There is a now PR up for this.

**Comment by GiorgiBeriashvili** [comment]
> Dio now has a `4.0.0-beta2` available with null-safety. It is no longer compatible.

Very nice!

> There is a now PR up for this.

I will take a look, thank you!

**Comment by AndreGroeschel** [comment]
Hi there,

is there any update on the progress for supporting null safety? We would like to upgrade to the latest Dio and are blocked by mock adapter right now. Any way we can help?

**Comment by GiorgiBeriashvili** [comment]
> Hi there,
> 
> is there any update on the progress for supporting null safety? We would like to upgrade to the latest Dio and are blocked by mock adapter right now. Any way we can help?

Hey!

We had three PRs on hold due to reviewer inactivity, but that issue has been resolved as of now. #83 will have to make slight changes mostly due to merge conflicts and then it will be merged, hopefully very soon.

Thank you for your patience and offering of help!

**Comment by kuhnroyal** [comment]
Can we please get a last release prior to the null-safety merge, basically the current develop. That would help a lot during the migration to the 0.2.0 version.

**Comment by GiorgiBeriashvili** [comment]
> Can we please get a last release prior to the null-safety merge, basically the current develop. That would help a lot during the migration to the 0.2.0 version.

[It](https://github.com/lomsa-dev/http-mock-adapter/pull/88) will be done soon!

#83 will most likely follow pretty soon as well. Thank you for your patience! :heart:

**Comment by GiorgiBeriashvili** [comment]
Version `0.2.0` and upwards supports null-safety, thus this issue can be considered closed.
